qemu/tests/migration.py

```diff
@@ -206,17 +209,29 @@ def check_dma():
                     guest_stress_deamon, ())
                 deamon_thread.start()
 
+            capabilities = ast.literal_eval(params.get("migrate_capabilities", "{}"))
+            inner_funcs = ast.literal_eval(params.get("migrate_inner_funcs", "[]"))
+
             # Migrate the VM
             ping_pong = params.get("ping_pong", 1)
             for i in range(int(ping_pong)):
                 if i % 2 == 0:
                     logging.info("Round %s ping..." % str(i / 2))
                 else:
                     logging.info("Round %s pong..." % str(i / 2))
-                vm.migrate(mig_timeout, mig_protocol, mig_cancel_delay,
-                           offline, check,
-                           migration_exec_cmd_src=mig_exec_cmd_src,
-                           migration_exec_cmd_dst=mig_exec_cmd_dst, env=env)
+                try:
+                    vm.migrate(mig_timeout, mig_protocol, mig_cancel_delay,
+                               offline, check,
+                               migration_exec_cmd_src=mig_exec_cmd_src,
+                               migration_exec_cmd_dst=mig_exec_cmd_dst,
+                               migrate_capabilities=capabilities,
+                               mig_inner_funcs=inner_funcs,
+                               env=env)
+                except qemu_monitor.MonitorNotSupportedMigCapError as e:
+                    logging.info("Unable to access capability: %s" % e)
+                    raise exceptions.TestSkipError(e)
```

Comment:
we have stopped using `exceptions` instead `test.cancel()` would help us here.

Reply:
OK, see if that new version is better.